### PR TITLE
chore: Order initialized nodes for Consolidation

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

Orders the `existingNodes` in a consistent manner in the scheduling algorithm. This ensures that:
1. Existing nodes that are being simulated against will have a consistent ordering so that the nomination mechanism should often pick the same node on multiple scheduling simulations (making PodNominations more consistent)
2. Initialized nodes will be tried first in consolidation so that it's less likely for consolidation to fail by scheduling a pod that it is trying to consolidate against an un-initialized node

**How was this change tested?**

`make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/test/expectations/expectations.go [hunk 1]
```diff
@@ -302,21 +302,32 @@ func ExpectProvisionedNoBindingWithOffset(offset int, ctx context.Context, c cli
 	return bindings
 }
 
-func ExpectMachineDeployed(ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, machine *v1alpha5.Machine) (*v1alpha5.Machine, *v1.Node) {
-	return ExpectMachineDeployedWithOffset(1, ctx, c, cluster, cloudProvider, machine)
+func ExpectMachineDeployedNoNode(ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, m *v1alpha5.Machine) *v1alpha5.Machine {
+	return ExpectMachineDeployedNoNodeWithOffset(1, ctx, c, cluster, cloudProvider, m)
 }
```
Comment (ellistarn) [inline pkg/test/expectations/expectations.go:307]: Why bother with this if it's just a passthrough?

Comment (jonathan-innis) [reply pkg/test/expectations/expectations.go:307]: This is common in the Ginkgo library. Happy to think about how we can refactor it in the future to make this more DRY but the caller not having to reason about offsets is generally a good way to go about things

File: pkg/controllers/provisioning/scheduling/scheduler.go [hunk 2]
```diff
@@ -288,6 +288,12 @@ func (s *Scheduler) calculateExistingMachines(stateNodes []*state.StateNode, dae
 			s.remainingResources[node.Labels()[v1alpha5.ProvisionerNameLabelKey]] = resources.Subtract(s.remainingResources[node.Labels()[v1alpha5.ProvisionerNameLabelKey]], node.Capacity())
 		}
 	}
+	// Order the existing nodes for scheduling with initialized nodes first
+	// This is done specifically for consolidation where we want to make sure we schedule to initialized nodes
+	// before we attempt to schedule un-initialized ones
+	sort.SliceStable(s.existingNodes, func(i, j int) bool {
+		return s.existingNodes[i].Initialized()
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply pkg/controllers/provisioning/scheduling/scheduler.go]: Updated to conform with the `Less()` method. Good callout! 🎉

**Review by tzneal — APPROVED** [review]
lgtm

**Pull request merged**